sympy/solvers/solveset.py

```diff
@@ -1948,6 +2157,12 @@ def add_type(lhs, rhs, symbol, domain):
         # check if it is logarithmic type equation
         elif _is_logarithmic(lhs, symbol):
             result = _solve_logarithm(lhs, rhs, symbol, domain)
+        elif _is_lambert(lhs, symbol):
+            # try to get solutions in form of Lambert
+            result = _solve_as_lambert(lhs, rhs, symbol, domain)
+            if isinstance(result, ConditionSet):
+                if _is_bivariate(lhs, symbol):
+                    result = _solve_as_bivariate(lhs, rhs, symbol, domain)
```

Comment:
Why is `solve_as_bivariate` separate from `solve_as_lambert`?

Reply:
I did not get your point, 
Actually according to the logic of code `solve_as_bivariate` is a subset of `_is_lambert` ( (`L-2160`) that's why it's under `elif _is_lambert(lhs, symbol):`   (`L-2160`)  ), so I think it is not separate but it's trying to work in different way ( see  `_solve_as_bivariate ` function) ....

Reply:
Btw, this code is written in https://github.com/sympy/sympy/pull/14972 ...